controllers/users.js

```diff
@@ -0,0 +1,107 @@
+const Users = require("../models/userSchema");
+const Joi = require("joi");
+const jwt = require("jsonwebtoken");
+
+const schemaLogin = Joi.object({
+    password: Joi.string().min(10).required(),
+    email: Joi.string().email().required(),
+    subscription: Joi.string().valid("starter", "pro", "business").default("starter"),
+    token: Joi.string().default(null),
+  });
+
+const register = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  const { error } = schemaLogin.validate({email, password });
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+
+    try {
+        const user = await Users.findOne({ email }).lean();
+
+        if (user) {
+            return res.status(409).json({ message: 'This email is already taken' });
+        }
+
+        const newUser = new Users({ email });
+        await newUser.setPassword(password);
+        await newUser.save();
+
+      return res.status(201).json(
+        {
+          "user": {
+            "email": email,
+            "subscription": "starter",
+            }
+        });
+    } catch (error) {
+        next(error);
+    }
+};
+
+const login = async (req, res, next) => {
+  
+  const { email, password } = req.body;
+  const user = await Users.findOne({ email })
+  
+  const { error } = schemaLogin.validate({ email, password });
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+  
+  if (!user){
+    return res.status(401).json({message: "User not found"})
+  }
+
+  const isPasswordCorrect = await user.validatePassword(password)
+
+  if (isPasswordCorrect) {
+      const payload = { id: user._id, }
+    const token = jwt.sign(
```

Comment:
Coś jest nie tak z indentacją, widzisz?

Reply:
Wiem, że nie działa mi to najlepiej, ale nie za bardzo potrafię dojść do tego co tu zepsułem.